flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/src/main/java/org/apache/flink/cdc/connectors/kafka/json/canal/CanalJsonSerializationSchema.java

```diff
@@ -132,6 +132,17 @@ public byte[] serialize(Event event) {
         }
 
         DataChangeEvent dataChangeEvent = (DataChangeEvent) event;
+        reuseGenericRowData.setField(
+                3, StringData.fromString(dataChangeEvent.tableId().getSchemaName()));
```

Comment:
`org.apache.flink.table.data.StringData#fromString` was used to generate binary data from string here, but I noticed that `org.apache.flink.cdc.common.data.binary.BinaryStringData#fromString` is more frequently used in CDC code base. Though they're basically the same (CDC version was copied from Flink), is it better if we can stick to one consistent binary encoding algorithm?

Reply:
reuseGenericRowData will be serialized in [SerializationSchema](https://github.com/apache/flink/blob/c7a50d05186ca0fd02b2a54394b9ad522d10e62f/flink-core/src/main/java/org/apache/flink/api/common/serialization/SerializationSchema.java#L54), so Flink types were passed here.

Reply:
Got it, thanks for clarification.